src/ssa.c

```diff
@@ -0,0 +1,1152 @@
+void bb_forward_traversal(fn_t *fn,
```

Comment:
How about defining a structure which contains `fn`, `bb`, and two callbacks? Therefore, we can operate with an explicit pointer without passing these parameters.